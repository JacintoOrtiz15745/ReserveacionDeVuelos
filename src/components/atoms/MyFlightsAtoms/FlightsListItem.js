import React from 'react';
import {View} from 'react-native';
import {styles} from '../../../styles/MyFlightsStyles';
import GenericListItem from './GenericFlightListItem';
import {FlightsListConstants} from '../../../utils/Constants';

const FlightsListItem = ({item}) => {
  return (
    <View style={styles.fatherContainer} testID="view">
      <GenericListItem
        ContainerStyle={styles.TopContainer}
        Title={item.originCity}
        Subtitle={item.destinyCity}
        TitleFont={styles.country}
        SubtitleFont={styles.country}
        Icon={FlightsListConstants.planeIconName}
      />
      <GenericListItem
        ContainerStyle={styles.middleContainer}
        Title={item.originCountry}
        Subtitle={item.destinyCountry}
        TitleFont={styles.city}
        SubtitleFont={styles.city}
      />
      <GenericListItem
        ContainerStyle={styles.bottomContainer}
        Title={item.date}
        Subtitle={`${item.passengers} ${FlightsListConstants.passengers}`}
        TitleFont={styles.dateTitle}
        SubtitleFont={styles.passengersTitle}
      />
    </View>
  );
};

export default FlightsListItem;
