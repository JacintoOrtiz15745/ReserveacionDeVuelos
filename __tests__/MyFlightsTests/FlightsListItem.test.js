import React from 'react';
import {shallow} from 'enzyme';
import FlightsListItem from '../../src/components/atoms/MyFlightsAtoms/FlightsListItem';
const item = {
  id: '1',
  originCountry: 'BEG',
  destinyCountry: 'AMS',
  originCity: 'Serbia',
  destinyCity: 'Netherlands',
  date: 'September 3, 2021',
  passengers: 5,
};

const props = {
  item,
};

describe('Render correctly', () => {
  const wrapper = shallow(<FlightsListItem {...props} />);
  test('snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('Render correctly', () => {
    const render = wrapper.findWhere((node) => node.prop('testID') === 'view');
    expect(render.exists()).toBeTruthy();
  });
  test('Render correctly', () => {
    const render = wrapper.findWhere((node) => node.prop('testID') === 'view');
    expect(render.props().children.length).toBe(3);
  });
  
});
