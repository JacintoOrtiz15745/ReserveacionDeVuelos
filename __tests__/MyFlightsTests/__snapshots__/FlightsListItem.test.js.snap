// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly snapshot 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 2,
      "marginBottom": 15,
    }
  }
  testID="view"
>
  <GenericListItem
    ContainerStyle={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
    Icon="plane"
    Subtitle="AMS"
    SubtitleFont={
      Object {
        "fontSize": 30,
        "fontWeight": "bold",
      }
    }
    Title="BEG"
    TitleFont={
      Object {
        "fontSize": 30,
        "fontWeight": "bold",
      }
    }
  />
  <GenericListItem
    ContainerStyle={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#b6b7ba",
        "borderBottomWidth": 2,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "marginTop": 5,
      }
    }
    Subtitle="Netherlands"
    SubtitleFont={
      Object {
        "color": undefined,
      }
    }
    Title="Serbia"
    TitleFont={
      Object {
        "color": undefined,
      }
    }
  />
  <GenericListItem
    ContainerStyle={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
      }
    }
    Subtitle="5 passengers"
    SubtitleFont={
      Object {
        "fontWeight": "bold",
      }
    }
    Title="September 3, 2021"
    TitleFont={
      Object {
        "fontWeight": "bold",
      }
    }
  />
</View>
`;
